app/Domain/User/Repository/UserRepository.php

```diff
@@ -3,7 +3,9 @@
 
 use App\Domain\User\User;
 
-interface UserCreatorRepository
+interface UserRepository
 {
     public function insert(User $user): int;
+
+    public function findBy(string $property, string $valueProperty): User | null;
```

Comment:
```suggestion
    public function findBy(string $property, string $valueProperty): ?User;
```

Agora é permitida esse tipo de sintaxe para representar o `null`